apps/opik-backend/src/main/java/com/comet/opik/domain/SpanDAO.java

```diff
@@ -444,6 +493,102 @@ public Mono<Void> insert(@NonNull Span span) {
                 .then();
     }
 
+    @Trace(dispatcher = true)
+    public Mono<Void> batchInsert(@NonNull List<Span> spans) {
+        return Mono.from(connectionFactory.create())
+                .flatMapMany(connection -> {
+                    if (spans.isEmpty()) {
+                        return Mono.just(0L);
+                    }
+
+                    return insert(spans, connection);
+                })
+                .then();
+    }
+
+    private Publisher<? extends Result> insert(Collection<Span> spans, Connection connection) {
+
+        return makeMonoContextAware((userName, workspaceName, workspaceId) -> {
+            List<TemplateUtils.QueryItem> queryItems = getQueryItemPlaceHolder(spans.size());
+
+            var template = new ST(PLAIN_INSERT)
+                    .add("items", queryItems);
+
+            var statement = connection.createStatement(template.render());
+
+            int i = 0;
+            for (Span span : spans) {
+
+                statement.bind("id" + i, span.id())
+                        .bind("project_id" + i, span.projectId())
+                        .bind("trace_id" + i, span.traceId())
+                        .bind("name" + i, span.name())
+                        .bind("type" + i, span.type().toString())
+                        .bind("start_time" + i, span.startTime().toString());
+
+                if (span.parentSpanId() != null) {
+                    statement.bind("parent_span_id" + i, span.parentSpanId());
+                } else {
+                    statement.bind("parent_span_id" + i, "");
+                }
```

Comment:
Probably many of these `else` statements are no longer necessary, you should probably revisit these and try to remove them, in order to simplify the logic of this method. ClickHouse explicit and implicit defaults should do the trick here.

Reply:
The issue here is R2DBC. It doesn't allow sending nulls in the bind function. We could use bindNull and rely on Clickhouse default, but else is needed anyway.